web/plugs/user_authorizer.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Janitor.Plugs.UserAuthorizer do
+  use Timex
+  import Plug.Conn
+
+  alias Janitor.User
+  alias Janitor.Repo
+
+  def init(_) do
+  end
+
+  def call(conn, _) do
+    perform_checks(
+      {conn, nil},
+      [
+        &extract_token_claims/1,
+        &check_expiration_time/1,
+        &assign_current_user/1
+      ]
+    )
+  end
+
+  #PRIVATE
+
+  defp perform_checks({conn, nil}, []), do: end_processing(conn, nil)
+  defp perform_checks({conn, nil}, [check_fun | remaining_checks]) do
+    perform_checks(check_fun.(conn), remaining_checks)
+  end
+  defp perform_checks({conn, err}, _), do: end_processing(conn, err)
+
+  defp end_processing(conn, err) do
```

Comment:
To be fair this function is useless. These cases could be handled directly in cases 1 and 3 of the `perform_checks` function:

```elixir
defp perform_checks({conn, nil}, []) do: conn
# ...
defp perform_checks({conn, err}, _), do
  conn |> send_resp(403, "Unauthorized request") |> Plug.Conn.halt
end
```